### PR TITLE
add cli

### PR DESCRIPTION
**Pull request by jrhizor**
- adds `stoat init`
- adds `stoat local`

Still working on adding tests. Docs will go in a separate PR. I think it would also make sense to separately move all of the action-related code into its own subdir so things like Jest don't get confused by looking at subdirectories.

https://www.npmjs.com/package/stoat

Until this is merged, I don't want to keep on adding versions so it's probably best to run `npm run build && npm i -g` in the `cli` directory if you want to test on your own computer.

### CODE REVIEW

File: cli/src/lib/init/stoatActionHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        const workflow: any = yaml.load(contents);
+
+        if (('jobs' in workflow && 'on' in workflow && 'pull_request' in workflow.on) || 'push' in workflow.on) {
+          for (const job in workflow.jobs) {
+            if ('steps' in workflow.jobs[job]) {
+              const steps = workflow.jobs[job].steps as any[];
+
+              if (
+                steps.length > 0 &&
+                steps.filter((step) => 'uses' in step && step.uses.includes('stoat-dev/stoat-action')).length === 0
```
Comment (tuliren) [inline cli/src/lib/init/stoatActionHelpers.ts:70]: Nitpick. The `Array#some` method tests whether there is at least one element that satisfies the condition. Using this method eliminates the necessity to manually check the length.

```suggestion
                !steps.some((step) => 'uses' in step && step.uses.includes('stoat-dev/stoat-action'))
```

File: cli/src/lib/local/portHelper.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import chalk from 'chalk';
+import portfinder from 'portfinder';
+
+// s+t+o+a+t = 19+20+15+1+20 = 75
+export const DEFAULT_PORT = 8075;
```
Comment (tuliren) [inline cli/src/lib/local/portHelper.ts:5]: 😄

File: cli/package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "lint": "eslint src/**/*.ts",
+    "create": "npm run build && npm run test",
+    "build": "npm run format && ncc build --source-map --license licenses.txt",
+    "local": "npm i -g && stoat",
+    "refresh": "rm -rf ./node_modules ./package-lock.json && npm install",
+    "test": "jest --coverage"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/stoat-dev/stoat.git"
```
Comment (tuliren) [inline cli/package.json:22]:
```suggestion
    "url": "git+https://github.com/stoat-dev/stoat-action.git"
```

`stoat` is private.

**Review by tuliren — APPROVED** [review]
Marvelous!

File: cli/src/lib/init/stoatActionHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  if (relevantJobs.length > 0) {
+    const jobList = relevantJobs.map((job) => `   - ${job.workflowFile} (${job.name})`).join('\n');
+
+    const questions = [
+      {
+        type: 'list',
+        name: 'action',
+        message: `\nThe following GitHub jobs do not have Stoat actions:\n${jobList}\n\nThe action is necessary to publish build artifacts or metrics to Stoat.\n\nWould you like to add the Stoat GitHub action as the final step to these workflows?`,
+        choices: ['No (to all)', 'Yes (to all)', 'Approve each workflow individually'],
```
Comment (tuliren) [inline cli/src/lib/init/stoatActionHelpers.ts:97]: Nitpick. Maybe move `No (to all)` to the end? We do want users to add our action, and sometimes user may just choose the first option.

**Comment by tuliren** [comment]
This is really exciting!

**Comment by tuliren** [comment]
> I think it would also make sense to separately move all of the action-related code into its own subdir

Yes, definitely make sense. We can also use the same eslint and prettier to manage all three modules. Will probably need some monorepo management library.

File: cli/package.json (hunk 3 above)
Comment (jrhizor) [reply cli/package.json:22]: I copied this from the root action package.json, will fix there too

**Pull request merged**